File: easy/age_distribution.clj
(defn age [a]
  (cond
    (or (< a 0) (> a 100)) "This program is for humans"
    (< a 3) "Still in Mama's arms"
    (< a 5) "Preschool Maniac"
    (< a 12) "Elementary school"
    (< a 15) "Middle school"
    (< a 19) "High school"
    (< a 23) "College"
    (< a 66) "Working for the man"
    :else "The Golden Years"))

(with-open [rdr (clojure.java.io/reader (first *command-line-args*))]
  (doseq [line (line-seq rdr)]
    (println (age (Integer/parseInt line)))))

File: easy/age_distribution.r
categories <- c("This program is for humans", "Still in Mama's arms", "Preschool Maniac", "Elementary school", "Middle school", "High school", "College", "Working for the man", "The Golden Years")
ages <- c(0, 3, 5, 12, 15, 19, 23, 66, 101)

cat(sapply(as.integer(readLines(tail(commandArgs(), n=1))), function(s) {
  i <- 1
  while (i <= length(ages) && s >= ages[i]) {
    i <- i + 1
  }
  if (i > length(ages)) {
    i <- 1
  }
  categories[i]
}), sep="\n")
